packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -191,17 +196,24 @@ function createHeadInsertionTransformStream(
       }
 
       const insertion = await insert()
+      const encodedInsertion = encoder.encode(insertion)
       if (inserted) {
-        controller.enqueue(encoder.encode(insertion))
+        controller.enqueue(encodedInsertion)
         controller.enqueue(chunk)
         freezing = true
       } else {
-        const content = decoder.decode(chunk)
-        const index = content.indexOf('</head>')
+        const index = indexOfUint8Array(chunk, ENCODED_TAGS.CLOSED.HEAD)
```

Comment:
Hmm good call out. I see where you're coming from, plus I remember the code you had written where it scans for one of the first couple bytes we actually care about and then it does the tree traversal. IMO since you, Wyatt, or I would be the ones to make such a change in the future, maybe we leave a comment and iterate?

Reply:
yeah works for me